src/Microsoft.DotNet.Build.CloudTestTasks/PackageFiles/CloudTest.Helix.targets

```diff
@@ -0,0 +1,305 @@
+      </LocalPayloadFileForUpload>
+    </ItemGroup>
+
+    <!-- Debug output of work items, and a warning if there are none. -->
+    <Message Text="Files for upload :: @(LocalFileForUpload)" Importance="Low" />
+
+    <!-- verify the test archives were created -->
+    <Warning Condition="'@(LocalFileForUpload->Count())' == '0'" Text="Didn't find any archives in supplied HelixWorkItem(s)." />
+
+    <!-- Work Item payloads -->
+    <UploadToAzure
+      Condition="'@(LocalFileForUpload->Count())' != '0'"
+      ConnectionString="$(CloudDropConnectionString)"
+      ContainerName="$(ContainerName)"
+      Items="@(LocalFileForUpload->Distinct())"
+      Overwrite="$(OverwriteOnUpload)" />
+
+    <!-- Correlations payloads -->
+    <UploadToAzure
+      Condition="'@(LocalPayloadFileForUpload->Count())' != '0'"
+      ConnectionString="$(CloudDropConnectionString)"
+      ContainerName="$(ContainerName)"
+      Items="@(LocalPayloadFileForUpload->Distinct())"
+      Overwrite="$(OverwriteOnUpload)" />
+
+    <!-- Supplemental payload.  TODO: Can be combined with Correlation Payload items, there's not much(any?) difference -->
+    <UploadToAzure
+      ConnectionString="$(CloudDropConnectionString)"
+      ContainerName="$(ContainerName)"
+      Items="@(SupplementalPayload)"
+      Overwrite="$(OverwriteOnUpload)"
+      Condition="'@(SupplementalPayload)' != ''" />
+
+  </Target>
+
+  <Target Name="CreateTestListJson">
+    <!-- We always bring the supplemental payload folder.  
+         This contains whatever's in src\Microsoft.DotNet.Build.CloudTestTasks\RunnerScripts -->
+    <ItemGroup>
+      <CorrelationPayloadUri Include="@(SupplementalPayload->'$(DropUri)%(RelativeBlobPath)$(DropUriReadOnlyToken)')" />
+      <CorrelationPayloadUri Include="@(LocalPayloadFileForUpload->'$(DropUri)%(RelativeBlobPath)$(DropUriReadOnlyToken)')" />
+    </ItemGroup>
+    <PropertyGroup>
+      <!-- flatten it into a property as msbuild chokes on @(CorrelationPayloadUri) in FunctionalTest.CorrelationPayloadUris -->
+      <CorrelationPayloadUris>@(CorrelationPayloadUri)</CorrelationPayloadUris>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <HelixProcessedWorkItem>
+        <Command>%(Command)</Command>
+        <CorrelationPayloadUris>[$(CorrelationPayloadUris)]</CorrelationPayloadUris>
+        <PayloadUri>$(DropUri)%(RelativeBlobPath)$(DropUriReadOnlyToken)</PayloadUri>
+        <WorkItemId>%(WorkItemId)</WorkItemId>
+        <TimeoutInSeconds>%(TimeoutInSeconds)</TimeoutInSeconds>
+      </HelixProcessedWorkItem>
+    </ItemGroup>
+
+    <!-- I'd love a built-in way to do this, but I'm compromising here since this lets us use 
+         arbitrary extra metadata in the future to compose work items, then strip out extra stuff used along the way.
+         It lets us use a single item to bring along everything we might possibly need in the future -->
+    <RemoveItemMetadata Items="@(HelixProcessedWorkItem)" FieldsToRemove="RelativeBlobPath;PayloadFile">
```

Comment:
I guess this works as well but it seems like a big hammer just to clear out these properties.